addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js

```diff
@@ -3284,6 +3284,27 @@ X[]
                 },
             });
         });
+        it('should return an image in a parent selection', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<div id="parent-element-to-select"><img/></div>',
+                stepFunction: editor => {
+                    const sel = editor.document.getSelection();
+                    const range = editor.document.createRange();
+                    const parent = editor.document.querySelector('div#parent-element-to-select');
+                    range.setStart(parent, 0);
+                    range.setEnd(parent, 1);
+                    sel.removeAllRanges();
+                    sel.addRange(range);
+                    window.chai
+                        .expect(
+                            getTraversedNodes(editor.editable).map(node =>
+                                node.nodeType === Node.TEXT_NODE ? node.textContent : node.nodeName,
+                            ),
+                        )
+                        .to.eql(['DIV', 'IMG']);
```

Comment:
Well I'm not sure the div is even traversed per se to be honest. Would some other tests break if we change `getTraversedNodes` accordingly ?

Reply:
Yes, 47 of them, mainly collapsed selections in text nodes.